### PR TITLE
WT-9225 Generate call log for session.begin_transaction

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

**Comment by Siddhartha8899** [comment]
For reference, below is the call log generated for `wiredtiger_open`,  `open_session`,  `set_timestamp`, `query_timestamp`, `begin_transaction` API calls.

```
[
{
    "class_name" : "global",
    "method_name" : "wiredtiger_open",
    "connection_id": "0x11ee430",
    "input" : {
    },
    "output" : {
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{
    "class_name" : "connection",
    "method_name" : "open_session",
    "session_id": "0x1216710",
    "input" : {
    },
    "output" : {
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{
    "class_name" : "connection",
    "method_name" : "set_timestamp",
    "connection_id": "0x11ee430",
    "input" : {
        "config": "oldest_timestamp=1,stable_timestamp=1"
    },
    "output" : {
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{
    "class_name" : "connection",
    "method_name" : "query_timestamp",
    "connection_id": "0x11ee430",
    "input" : {
        "config": "get=oldest_timestamp"
    },
    "output" : {
        "timestamp_queried": "1"
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{
    "class_name" : "session",
    "method_name" : "begin_transaction",
    "session_id": "0x1216710",
    "input" : {
        "config": "read_timestamp=1"
    },
    "output" : {
    },
    "return" : {
        "return_val" : 0,
        "error_message" : ""
    }
},
{}]
```

File: src/call_log/call_log.c [hunk 1]
```diff
@@ -299,4 +299,33 @@ __wt_call_log_query_timestamp(
...
+    conn = S2C(session);
+
+    WT_RET(__call_log_print_start(session, "session", "begin_transaction"));
+    WT_RET(__wt_fprintf(session, conn->call_log_fst, "    \"session_id\": \"%p\",\n", session));
+
+    /*
+     * The begin transaction entry includes the timestamp configuration string which is copied from
+     * the original API call.
+     */
+    WT_RET(__wt_snprintf(config_buf, sizeof(config_buf), "\"config\": \"%s\"", config));
```
Comment (Sean04) [inline src/call_log/call_log.c:321]: This doesn't seem like a problem to me but something to consider is that the output appears as:

```
    "input" : {
        "config": "(null)"
    },
```

when a timestamp isn't passed in to begin_transaction.

Comment (Siddhartha8899) [reply src/call_log/call_log.c:321]: Good find @Sean04

Comment (Siddhartha8899) [reply src/call_log/call_log.c:321]: We could consider this scenario in the simulator application.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**